app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java

```diff
@@ -223,27 +211,78 @@ private void setMapStyle(GoogleMap googleMap) {
         }
     }
 
+
+    /**
+     * The purpose of this method is display the polygon on the map and
+     * call the right method for onClick polygon or on click the marker.
+     *
+     * @param map is the map to be used in our application
+     */
+    private void setupPolygons(GoogleMap map) {
+        mLayer = mViewModel.loadPolygons(map, getContext());
+        mLayer.addLayerToMap();
+
+        setupPolygonClickListener();
+        setupBuildingMarkerClickListener(map);
+        setupZoomListener(map);
+        classRoomCoordinateTool(map);
+    }
+
+    private void setupZoomListener(final GoogleMap map) {
+        map.setOnCameraMoveListener(() -> {
+            if (map.getCameraPosition().zoom > 19) {
+                mLayer.removeLayerFromMap();
+            } else {
+                mLayer.addLayerToMap();
+            }
+        });
+    }
```

Comment:
can these be moved to the drawpolygons class?

Reply:
It is possible however, it will lead to create a big drawpolygons class. Which is something i am trying to avoid